src/Bicep.Core.IntegrationTests/Scenarios/NameofFunctionTests.cs

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Bicep.Core.UnitTests.Assertions;
+using Bicep.Core.UnitTests.Utils;
+using FluentAssertions.Execution;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Bicep.Core.IntegrationTests.Scenarios
+{
+    [TestClass]
+    public class NameofFunctionTests
+    {
+        [TestMethod]
+        public void NameofFunction_OnObjectProperty_ReturnsPropertyName()
+        {
+            var result = CompilationHelper.Compile("""
+var obj = {
+  prop: 'value'
+}
+var name = nameof(obj.prop)
+""");
+
+            using (new AssertionScope())
+            {
+                result.Should().NotHaveAnyCompilationBlockingDiagnostics();
+                result.Template.Should().HaveValueAtPath("$.variables['name']", "prop");
+            }
+        }
+
+        [TestMethod]
+        public void NameofFunction_OnVariable_ReturnsVariableName()
+        {
+            var result = CompilationHelper.Compile("""
+var obj = {
+  prop: 'value'
+}
+var name = nameof(obj)
+""");
+
+            using (new AssertionScope())
+            {
+                result.Should().NotHaveAnyCompilationBlockingDiagnostics();
+                result.Template.Should().HaveValueAtPath("$.variables['name']", "obj");
+            }
+        }
+
+        [TestMethod]
+        public void NameofFunction_OnResource_ReturnsResourceSymbolicName()
+        {
+            var result = CompilationHelper.Compile("""
+resource myStorage 'Microsoft.Storage/storageAccounts@2019-06-01' = {
+  name: 'storage123'
+}
+
+var name = nameof(myStorage)
```

Comment:
this fails as the name variable is marked for inlining while it should be produced.

Reply:
Do you know if the `ReturnResultBuilder` delegate is failing due to inlining, or is it the `Evaluator` delegate?

Reply:
as I wrote on the chat, now I'm thinking that perhaps it does not matter if an unused variable if not present in the output template 🤔 maybe eventually we will do some tree shaking to remove such unnecessary code so it would not work anyway. probably if I change `var name` to `output name string` this will work.